squidpy/tl/_var_by_distance.py

```diff
@@ -0,0 +1,279 @@
+from __future__ import annotations
+
+from functools import reduce
+from itertools import product
+from typing import Any, Dict, List, Tuple, Union  # noqa: F401
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from scanpy import logging as logg
+from sklearn.metrics import DistanceMetric
+from sklearn.neighbors import KDTree
+from sklearn.preprocessing import MinMaxScaler
+
+from squidpy._docs import d
+from squidpy._utils import NDArrayA
+from squidpy.gr._utils import _save_data
+
+__all__ = ["var_by_distance"]
+
+
+@d.dedent
+def var_by_distance(
+    adata: AnnData,
+    groups: str | List[str] | NDArrayA,
+    cluster_key: str,
+    library_key: str | None = None,
+    design_matrix_key: str = "design_matrix",
+    covariates: str | List[str] | None = None,
+    metric: str = "euclidean",
+    spatial_key: str = "spatial",
+    copy: bool = False,
+) -> AnnData:
+    """
+    Build a design matrix consisting of gene expression by distance to selected anchor point(s).
+
+    Parameters
+    ----------
+    adata
+        Annotated data matrix.
+    groups
+        Anchor points to calculate distances from, can be a single gene,
+        a list of genes or a set of coordinates.
+    cluster_key
+        Annotation column in `.obs` to take anchor point(s) from.
+    library_key
+        specify slide location in '.obs' which contain identical anchor points.
+    design_matrix_key
+        Name of the design matrix saved to `.obsm`, defaults to "design_matrix".
```

Comment:
```suggestion
        Name of the design matrix saved to `.obsm`.
```

Reply:
the default is already in the function signature, no need to spell it out in docstrings.